### PR TITLE
Adds support for the GetInstalledPackageResourceRefs grpc call for fluxv2

### PR DESCRIPTION
**Pull request by absoludity**
Signed-off-by: Michael Nelson <minelson@vmware.com>

### Description of the change

This PR just adds support for the GetInstalledPackageResourceRefs call to the fluxv2 plugin.

The code is the same as that in the helm plugin (longer term we may want to have some shared functionality between the helm and fluxv2 plugins given they both work with flux).

I had to update the tests to work in the same way as the flux model.

I haven't done the in-real-life test (but will do), but this should ensure that we can now view the details of an installed package (and it will watch/track the resources)

### Benefits

Installed package details for flux can be viewed correctly.

### Applicable issues

- ref #3779

### CODE REVIEW

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 1]
```diff
@@ -32,9 +34,9 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
```
Comment (gfichtenholt) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:37]: I was a little concerned that k8s.io/apimachinery/pkg/util/yaml.Unmarshal might not be doing the same as sigs.k8s.io/yaml.Unmarshal (I vaguely remember some trouble like this with YAML or JSON Unmarshaller in the past) but after checking the code it looks like the former calls the latter, so should be ok

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 2]
```diff
@@ -673,3 +675,53 @@ func newFluxHelmRelease(chart *models.Chart, targetName types.NamespacedName, ve
...
+			Kind:       doc.Kind,
+			Name:       doc.Metadata.Name,
+			Namespace:  doc.Metadata.Namespace,
+		})
+	}
+
+	return refs, nil
+}
+
+type yamlMetadata struct {
```
Comment (gfichtenholt) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:717]: that and the next type feel like they belong in some pkg to be re-used by multiple components? There's nothing flux-specific about this

**Review by gfichtenholt — APPROVED** [review]
I am fine with this. I did post a question on the CNCF flux channel just to confirm whether this approach is the right one. If/when I find out otherwise, I will let you know. Thanks. Greg

**Review by gfichtenholt — COMMENTED** [review]
I did post a question on the CNCF flux channel just to confirm whether this approach is the right one. The main flux guy Stephan pointed me to this page: [https://fluxcd.io/docs/components/helm/helmreleases/#reconciliation](https://fluxcd.io/docs/components/helm/helmreleases/#reconciliation). In particular it says:
List all Kubernetes objects reconciled from a HelmRelease:

kubectl get all --all-namespaces
-l=helm.toolkit.fluxcd.io/name=""
-l=helm.toolkit.fluxcd.io/namespace=""

in other words, to look for certain labels. What do you think?

**Comment by absoludity** [comment]
> I did post a question on the CNCF flux channel just to confirm whether this approach is the right one. The main flux guy Stefan pointed me to this page: https://fluxcd.io/docs/components/helm/helmreleases/#reconciliation. In particular it says: List all Kubernetes objects reconciled from a HelmRelease:
> 
> ```
> kubectl get all --all-namespaces
> -l=helm.toolkit.fluxcd.io/name="<HelmRelease name>"
> -l=helm.toolkit.fluxcd.io/namespace="<HelmRelease namespace>"
> ```
> 
> in other words, to look for certain labels. What do you think?

Yeah, that approach is possible with or without flux, as  helm also adds labels to the resources in manages which can be used to query. Helm uses the [standard `app.kubernetes.io` labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels), so for example, in the context of the helm plugin itself, we could instead be doing the following to see all the resources managed by helm for the kubeapps installation with:

```
$ kubectl get all --all-namespaces -l app.kubernetes.io/managed-by=Helm -l app.kubernetes.io/instance=kubeapps
NAMESPACE   NAME                                                              READY   STATUS    RESTARTS   AGE
kubeapps    pod/kubeapps-9467885c4-xpmgf                                      2/2     Running   0          2d8h
kubeapps    pod/kubeapps-internal-apprepository-controller-78947f7bcc-pmghj   1/1     Running   0          2d8h
kubeapps    pod/kubeapps-internal-dashboard-8557f68747-gpnds                  1/1     Running   0          2d8h
kubeapps    pod/kubeapps-internal-kubeappsapis-56f97597f5-6fnn8               1/1     Running   0          2d8h
kubeapps    pod/kubeapps-internal-kubeops-86b95d45bb-mmd7n                    1/1     Running   0          2d8h
kubeapps    pod/kubeapps-postgresql-0                                         1/1     Running   0          9d

NAMESPACE   NAME                                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
kubeapps    service/kubeapps                         NodePort    10.96.3.247     <none>        80:30000/TCP   9d
kubeapps    service/kubeapps-internal-dashboard      ClusterIP   10.96.132.12    <none>        8080/TCP       9d
kubeapps    service/kubeapps-internal-kubeappsapis   ClusterIP   10.96.102.53    <none>        8080/TCP       9d
kubeapps    service/kubeapps-internal-kubeops        ClusterIP   10.96.118.207   <none>        8080/TCP       9d
kubeapps    service/kubeapps-postgresql              ClusterIP   10.96.233.95    <none>        5432/TCP       9d
kubeapps    service/kubeapps-postgresql-headless     ClusterIP   None            <none>        5432/TCP       9d

NAMESPACE   NAME                                                         READY   UP-TO-DATE   AVAILABLE   AGE
kubeapps    deployment.apps/kubeapps                                     1/1     1            1           9d
kubeapps    deployment.apps/kubeapps-internal-apprepository-controller   1/1     1            1           9d
kubeapps    deployment.apps/kubeapps-internal-dashboard                  1/1     1            1           9d

NAMESPACE   NAME                                                                    DESIRED   CURRENT   READY   AGE
kubeapps    replicaset.apps/kubeapps-656cb79454                                     0         0         0       9d
kubeapps    replicaset.apps/kubeapps-77bcd5fdb7                                     0         0         0       9d
kubeapps    replicaset.apps/kubeapps-9467885c4                                      1         1         1       2d8h
kubeapps    replicaset.apps/kubeapps-b967c57c8                                      0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-apprepository-controller-78947f7bcc   1         1         1       2d8h
kubeapps    replicaset.apps/kubeapps-internal-apprepository-controller-7c97c7f4cb   0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-apprepository-controller-f86d77c8d    0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-dashboard-69c4495686                  0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-dashboard-8557f68747                  1         1         1       2d8h
kubeapps    replicaset.apps/kubeapps-internal-dashboard-858bb8ccf7                  0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-56f97597f5               1         1         1       2d8h
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-648fcf57cd               0         0         0       7d2h
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-6559d565c9               0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-66d448cd9                0         0         0       8d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-69f5756944               0         0         0       7d1h
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-6b6b8ffdc4               0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-6bb67fc4f7               0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-6c5c8c5fc7               0         0         0       7d2h
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-7945456bc6               0         0         0       8d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-7db645c494               0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeappsapis-b4db987c5                0         0         0       7d2h
kubeapps    replicaset.apps/kubeapps-internal-kubeops-564d7658c5                    0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeops-655b48bd4c                    0         0         0       9d
kubeapps    replicaset.apps/kubeapps-internal-kubeops-86b95d45bb                    1         1         1       2d8h

NAMESPACE   NAME                                   READY   AGE
kubeapps    statefulset.apps/kubeapps-postgresql   1/1     9d
```

But doing so requires quite extensive RBAC that a Kubeapps user is not guaranteed to have. Even if you remove the `--all-namespaces`, it still requires RBAC to access everything encompassed by `all` in the given namespace. It may be safe to assume that for us (depending on other assumptions), but the current method for the helm plugin is guaranteed to result in a (smaller) list of resources that were created with the users credentials during the helm install without requiring any extra RBAC (smaller because it includes only the resources created by the installation, not those created by controllers as a result).

It is different here for fluxv2, given that the the installation is done with a chosen service account, but I've used the same method as used by helm for this initial implementation. I don't mind if we want to do something different for the fluxv2 plugin (but I probably won't have time to work on it right now). Happy to either park the PR for later, or add a TODO with info for considering later and move ahead, or if you want to grab it and continue... all options are fine with me.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go (hunk 2 above)
Comment (absoludity) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:717]: Yep, I mentioned in the description above that we may want to share functionality between the helm and flux plugins. Could be even broader than that.

**Review by gfichtenholt — APPROVED** [review]
Fair enough. I think a TODO with the link to your last comment will suffice. I am in the middle of 3 different things so won't be able to pick it up soon
